### ISSUE TITLE
[stgp demo] code need optimize, spend too much time/memory

### ISSUE THREAD

**Issue by zen-xingle** [IceLab-X/Mini-GP#14]
https://github.com/IceLab-X/Mini-GP/blob/a72779a0b4a178e987e663ed4893763633d9e82a/stgp_demo.py#L55

This line costs almost 98% time(70s) of the algorithm and occupies 8G memory. Optimize it or this algorithm is useless

**Comment by wayXing** [comment]
is sigma the whole covariance matrix? we do not actually need to compute the whole covariance matrix in practice.

**Comment by Fillip1233** [comment]
Yes, sigma is the whole covariance matrix. In older code versions, its large size resulted in significant time and memory overhead during computation. In the new version of the code, we have optimized the training code using the mathematical properties of the Kronecker product, and specific mathematical calculation formulas can be referred to [MultiTaskGP_IMC_MathTutorial2](https://github.com/IceLab-X/Mini-GP/blob/zhenjie's_workspace/MultiTaskGP_IMC_MathTutorial2.md)
